trunk-recorder/call.cc

```diff
@@ -121,6 +121,8 @@ void Call::end_call() {
     BOOST_LOG_TRIVIAL(info) << "[" << sys->get_short_name() << "]\tTG: " << this->get_talkgroup_display() << "\tFreq: " << FormatFreq(get_freq()) << "\tEnding Recorded Call - Last Update: " << this->since_last_update() << "s\tCall Elapsed: " << this->elapsed();
 
     final_length = recorder->get_current_length();
+    // Fixes https://github.com/robotastic/trunk-recorder/issues/103#issuecomment-284825841
+    start_time = stop_time - final_length;
```

Comment:
Interesting - let me dig into this. The Start time on trunking systems should be when the call goes on the control channel. Since the `final_length` excludes some of the dead area time, this way of calculating start_time maybe off.

Reply:
Ah, this might be something we only want to do on conventional channels then, with long-running recorders.

Reply:
I had done some changes to account for idle (dead air) time between calls on conventional and P25conventional calls
[https://github.com/gofaster/trunk-recorder/commit/1c502b3b5bca0f0ba365864ea3e1492fba67a048](url)
[https://github.com/gofaster/trunk-recorder/commit/e2122a0d2ae907ac0b19deef6567bd930bf6dd08](url)
[https://github.com/gofaster/trunk-recorder/commit/a3a8ba5635b8c89dc4b3ab9861d48280a04e1eb6](url)

Reply:
@gofaster thanks, that seems like the right approach. Your solution seems more robust so I'd prefer #322 to be merged and I can pull my more naive fix out.